sys/Makefile.dep

```diff
@@ -259,12 +259,6 @@ ifneq (,$(filter posix_sockets,$(USEMODULE)))
   USEMODULE += posix_headers
 endif
 
-ifneq (,$(filter shell_commands,$(USEMODULE)))
-  # shell_commands has been renamed to shell_cmds_default, but let's keep this
-  # for backward compatibility
-  USEMODULE += shell_cmds_default
-endif
```

Comment:
Is this really necessary?
This will break *all* external apps, keeping those few lines around wouldn't hurt and avoid the churn on users.

Reply:
This module is deprecated for more than a year already. We were supposed to remove it after 2023.07 release according to `makefiles/pseudomodules.inc.mk`
I guess we give enough time to our users to migrate already.

> This will break all external apps, keeping those few lines around wouldn't hurt and avoid the churn on users.

The question is do we want to keep backward compatibility forever ?

Reply:
I'm opting for removal. If your old app *really* needs it, you can still use old releases of RIOT.
(I also believe it is a good thing, to annoy your users to migrate to new APIs regularly, as long as it is done in a sensible manner. A bit like Apple does it. But that's clearly an personal opinion)

Reply:
I'm also in favor of removal, in the end that's what our yet-to-be-documented deprecation process is for.